doc/docs/Python_Tutorials/Custom_Source.md

```diff
@@ -304,7 +304,7 @@ if mp.am_master():
 
 There are three items to note in this script. (1) The line source spans the entire length of the cell in the $x$ direction (i.e., $L$ is `sx`). (2) The number of point dipoles in Method 2 is `sx*resolution`, one per pixel. (3) The source amplitude function in Method 3 is specified by the `amp_func` property of the [`Source`](../Python_User_Interface.md#source) object. In the case of a Fourier cosine series as conventionally written, $\cos (m\pi x)/L$ is defined over the interval $x=[0,L]$ such that $x=0$ corresponds to the *edge* of the source, not the center. Since the source region in this example is defined in $[-L/2,+L/2]$, the amplitude function must shift its $x$ coordinate argument by $+L/2$ or `0.5*sx`.
 
-Method 3 requires a convergence check in which $M$ (`nsrc` in the script) is repeatedly doubled until the change in the results are within a desired tolerance of e.g., < 1%. For this example, $M=15$ was found to be sufficient.
+Method 3 requires a convergence check in which $M$ (`nsrc` in the script) is repeatedly doubled until the change in the results are within a desired tolerance of e.g., < 1%. For this example, $M=11$ was found to be sufficient. Note that because a line source with a cosine amplitude function is analogous to generating a planewave at a discrete angle, at each frequency $\omega$ there exists a cutoff $M$ beyond which there are *no* propagating planewaves. The cutoff $M$ can be computed analytically using the grating equation: $\sqrt{\omega^2n^2 - \left(k_x+\frac{M\pi}{L}\right)^2} = 0$, where $n$ is the refractive index of the source medium and $k_x$ is the Bloch-periodic wavevector in the $x$ direction. In this example, $\omega=2\pi$, $L=1.5$, $k_x=0$, and $n=3.45$ for which the cutoff $M$ is 11.
```

Comment:
Technically, what happens is that for M > 11 the cosine source produces evanescent waves *in the material*, but it scatters into other Fourier components (including propagating waves) once it hits the grating.   So, it still produces propagating waves, but the amplitude of the propagating waves (and hence the contribution to the power) decreases exponentially for M > 11 (as the coupling between the source and the grating decreases).   (This only happens for periodic boundary conditions where the line source extends for the whole period.)

Reply:
Might be helpful to show the convergence plot with M

Reply:
Fixed.